Assets/HoloToolkit-UnitTests/Editor/Utilities/Extensions/WorldAnchorManagerExtensions.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using UnityEngine;
+
+namespace HoloToolkit.Unity.Tests.Extensions
+{
+    /// <summary>
+    /// This class extends the world anchor manager with methods constrained to test logic.
+    /// </summary>
+    public static class WorldAnchorManagerExtensions
+    {
+        internal static string GenerateAnchorName(this WorldAnchorManager anchorManager, GameObject gameObjectToAnchor, string anchorName = null)
+        {
+            return string.IsNullOrEmpty(anchorName) ? gameObjectToAnchor.name : anchorName;
```

Comment:
Did you only copy this logic and leave it in the anchor manager? If so, this method is only ever called in the tests, which doesn't really make sense. 
This definetly has to be in the main folder, not in `UnitTests`. There also is no need for this to be an extension, as the anchormanager isn't used. I would either put it in `WorldAnchorManager` directly or make a `WorldAnchorUtilities` and put it in there where both version would be `public static`. Like that you wouldn't need to initialise the `WorldAnchorManager` for the test as well.

In conclusion:
1. Move this logic out of the `AttachAnchor` Method
2. Make a public static Method in either WorldAnchorManager or WorldAnchorUtilities
3. Test that instead, which should make the setup a lot easier too.

Reply:
The logic is duplicated code. It is both in the test method and in the `WorldAnchorManager`, which is kind of bad. Do you think I should create a `WorldAnchorUtilities` in the main folder? This class would have only one method as of now... Probably it is better, if I take the method into the `WorldAnchorManager` directly and make it `public static`. I think, I was just afraid of changing too much 👎 .